src/app/work-item/work-item-list/work-item-list.component.ts

```diff
@@ -248,4 +248,45 @@ export class WorkItemListComponent implements OnInit, AfterViewInit {
     this.showTypesOptions = false;
     this.router.navigate(['/work-item-list/detail/new?' + type]);
   }
-}
\ No newline at end of file
+
+  onDragStart() {
+    console.log('on drag start');
+  }
+
+  //Event listener for WI drop.
+  onDragEnd(workItemId: number) {
+    let droppedElement = document.getElementById('work-item-'+workItemId);
+    let previousElement = droppedElement.previousElementSibling;
+    let nextElement = droppedElement.nextElementSibling;
+    let prevItemId: string;
+    let nextItemId: string;
+
+    if(previousElement) {
+      prevItemId = previousElement.getAttribute('data-id');
+    }
+    else {
+      prevItemId = '';
+    }
+    if(nextElement) {
+      nextItemId = nextElement.getAttribute('data-id');
+    }
+    else {
+      nextItemId = '';
+    }
+
+    this.workItemService
```

Comment:
@vikram-raj @sanbornsen is this the best place for this calculation and service call or should this logic reside in work-item-services?
list-component should send the data - work item id, updated work item with the previous and next work item ids.
Services should have a reorder function which 
1. accepts the work item to be updated
2. call the re-order url
3. update the index map

If we have a function like this it can be re-used.